uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-calendar.component.spec.js

```diff
@@ -0,0 +1,496 @@
+              return !period.current;
+            }).sample();
+            label = controller.labelPeriod(period);
+          });
+
+          it('returns the title as it is', function () {
+            expect(label).toBe(period.title);
+          });
+        });
+      });
+
+      describe('refresh()', function () {
+        describe('loading flag', function () {
+          beforeEach(function () {
+            controller.refresh();
+          });
+
+          it('does not mark the entire page as loading', function () {
+            expect(controller.loading.page).not.toBe(true);
+          });
+
+          it('marks the calendar content as loading', function () {
+            expect(controller.loading.calendar).toBe(true);
+          });
+        });
+
+        describe('source of refresh', function () {
+          var oldMonths, spyLoadContacts;
+
+          beforeEach(function () {
+            spyLoadContacts = spyOnSubCtrlLoadContacts();
+
+            compileComponent();
+            oldMonths = controller.months;
+
+            spyLoadContacts.calls.reset();
+            $rootScope.$emit.calls.reset();
+          });
+
+          describe('when the source of the refresh is a period change', function () {
+            beforeEach(function () {
+              controller.refresh('period');
+              digest(true);
+
+              simulateMonthsInjected(controller.months.length);
+            });
+
+            it('rebuilds the months structure', function () {
+              expect(controller.months).not.toBe(oldMonths);
+            });
+
+            it('does not reloads the contacts', function () {
+              expect(spyLoadContacts).not.toHaveBeenCalled();
+            });
+
+            it('sends the "show months" signal without forcing data reload', function () {
+              expect($rootScope.$emit).toHaveBeenCalledWith(
+                'LeaveCalendar::showMonths',
+                jasmine.any(Array),
+                false
+              );
+            });
+          });
+
+          describe('when the source of the refresh is a contact filters change', function () {
+            beforeEach(function () {
+              controller.refresh('contacts');
+              digest(true);
+
+              simulateMonthsInjected(controller.months.length);
+            });
+
+            it('does not rebuild the months structure', function () {
+              expect(controller.months).toBe(oldMonths);
+            });
+
+            it('reloads the contacts', function () {
+              expect(spyLoadContacts).toHaveBeenCalled();
+            });
+
+            it('sends the "show months" signal with forcing data reload', function () {
+              expect($rootScope.$emit).toHaveBeenCalledWith(
+                'LeaveCalendar::showMonths',
+                jasmine.any(Array),
+                true
+              );
+            });
+          });
+        });
+      });
+
+      function amend2016Period (params) {
```

Comment:
Please add description comment to those functions (except compileComponent that should be at the very end?)